kong/plugins/opentelemetry/handler.lua

```diff
@@ -142,11 +142,7 @@ local function process_span(span)
 
   local pb_span = transform_span(span)
 
-  local len = spans_queue.n or 0
-  len = len + 1
-
-  spans_queue[len] = pb_span
-  spans_queue.n = len
+  insert(spans_queue, pb_span)
```

Comment:
This might be slow as table length is actually relatively expensive to calculate in Lua. It is common to store the length of the array in `spans_queue[0]` and then do the previous logic instead.

Reply:
Thanks for noticing, used `BatchQueue` to replace the worker-level cache table. I have an idea that only the privileged worker handles the queue and sends requests; the other workers insert data into the queue through the new events library. That could reduce the performance impact on normal workers.